tools/scripts/generateSharedConstants.rb

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env ruby
+#
+# This file generates a static json file, and uses it to create enum-like objects
+# that can be used in both dashboard and apps. It does this by generating two
+# static files - one which is a ruby file to be used by dashboard, the other
+# which is a js file to be used by apps. In both cases, the expectation is that
+# the generated files get checked into the repo
+#
```

Comment:
Will this have to be run manually, or can we expect to add it to the build process?